examples/android/BUILD.bazel

```diff
@@ -18,6 +18,35 @@ cc_library(
     alwayslink = True,  # Required since JNI symbols appear to be unused
 )
 
+rust_library(
+    name = "rust_lib2",
+    srcs = ["demo2.rs"],
+    edition = "2018",
+)
+
+cc_library(
+    name = "jni_shim2",
+    srcs = ["android_link_hack.c"],  # Required because of https://github.com/bazelbuild/rules_rust/issues/1271
```

Comment:
this should only be in a single cc_library in the whole dependency tree

Reply:
It seems like I need to change the function name too? 

```
/usr/bin/ld.gold: error: bazel-out/k8-fastbuild-android-ST-b08757e62318/bin/_objs/jni_shim2/android_link_hack2.pic.o: multiple definition of 'call_link_hack'
/usr/bin/ld.gold: bazel-out/k8-fastbuild-android-ST-b08757e62318/bin/_objs/jni_shim/android_link_hack.pic.o: previous definition here
```

Reply:
So the build isn't happy if the two rust files (used in two different `rust_library`s) have any duplicate function signatures.

```
/usr/bin/ld.gold: error: bazel-out/k8-fastbuild-android-ST-b08757e62318/bin/librust_lib2-855537059.a(rust_lib2-855537059.rust_lib2.a90be88302ad5382-cgu.0.rcgu.o): multiple definition of 'print_something_from_rust'
/usr/bin/ld.gold: bazel-out/k8-fastbuild-android-ST-b08757e62318/bin/librust_lib-443237737.a(rust_lib-443237737.rust_lib.6f4c6aab038b4cfe-cgu.0.rcgu.o): previous definition here
```

This seems like an unintended restriction to be resolved.

Reply:
If there should only be one `cc_library` in the whole dependency tree, doesn't that mean there is only one `android_library` in the whole tree as well?

Reply:
i think the ideal is that every rust_library (any number) depends on the single cc_library containing the shim. if that doesn't work because of the linked `alwayslink` issue, then i think you'd have to change the names of everything in the "link hack" (and ofc rust symbols can't overlap just like normal)